server-protocol.md

```diff
@@ -104,6 +114,46 @@ was sent in the wrong order), an `error` response will be sent, This response
 will include the error string in the `error` key, and a full copy of the
 original message dictionary in `orig`.
 
+
+## Permission and Abilities
+
+Server operators may wish to deny service to some clients.
+
+One such use-case is if the server is under a Denial of Service (DoS)
+attack or other malicious activity.
+
+In the `welcome` message the server may include a `permission-required` key. If provided, this will point to a `dict` with one key per supported method along with any method-specific options.
+For example::
+
+    {
+        "none": {}
+        "hashcash": {
+            "bits": 6,
+            "resource": "arbitrary string"
+        }
+    }
+
+Currently, the following are supported by the protocol:
+
+* `none`: no permission required, send a normal `bind`.
+* `hashcash`: Includes keys for `bits` and `resource` used as input as per the
+[Hashcash](http://hashcash.org) specifications. The client must
+include a reply that conforms to the [Hashcash v1](http://hashcash.org/docs/hashcash.html#stamp_format__version_1_)
+format in the `submit-permissions` part of the `bind` message::
+
+    {
+        "hashcash": {
+            "stamp": "1:20:210610:arbitrary string::i9CmTZR6oGBx7kr2:06czd"
+        }
+    }
+
+The above stamp was generated with `hashcash -m "arbitrary string" -b 6`.
```

Comment:
The `hashcash` binary seem sensitive about the order of parameters. The `-b` has to come first and then `-m`. Otherwise, it by default produces a partial hash with 20 bits of zeros in the most significant bits.

Reply:
Even in the example above, "20" doesn't match the input "6", so I guess the one you used to generate them also has this "bug". (fwiw, I used the `hashcash` package that comes with Debian).

Reply:
nice catch .. fixed it up